packages/tracing/src/integrations/node/express.ts

```diff
@@ -211,3 +239,71 @@ function patchMiddleware(router: Router, method: Method): Router {
 function instrumentMiddlewares(router: Router, methods: Method[] = []): void {
   methods.forEach((method: Method) => patchMiddleware(router, method));
 }
+
+/**
+ * Patches the prototype of Express.Router to accumulate the resolved route
+ * if a layer instance's `match` function was called and it returned a successful match.
+ *
+ * @see https://github.com/expressjs/express/blob/master/lib/router/index.js
+ *
+ * @param appOrRouter the router instance which can either be an app (i.e. top-level) or a (nested) router.
+ */
+function instrumentRouter(appOrRouter: ExpressRouter): void {
+  // This is how we can distinguish between app and routers
+  const isApp = 'settings' in appOrRouter;
+
+  // In case the app's top-level router hasn't been initialized yet, we have to do it now
+  if (isApp && appOrRouter._router === undefined) {
+    appOrRouter.lazyrouter && appOrRouter.lazyrouter();
+  }
+
+  const router = isApp ? appOrRouter._router : appOrRouter;
+  const routerProto = Object.getPrototypeOf(router) as ExpressRouter;
+
+  const originalProcessParams = routerProto.process_params;
+  routerProto.process_params = function process_params(
+    layer: Layer,
+    called: unknown,
+    req: PatchedRequest,
+    res: ExpressResponse & SentryTracingResponse,
+    done: () => unknown,
+  ) {
+    // Base case: We're in the first part of the URL (thus we start with the root '/')
+    if (!req._reconstructedRoute) {
+      req._reconstructedRoute = '/';
+    }
+
+    // If the layers partial route has params, it's stored in layer.route, else it's in layer.path
```

Comment:
```suggestion
    // If the layer's partial route has params, it's stored in layer.route. Otherwise, it's in layer.path
```

What is "it" here? The route? I got confused reading this the first time thinking the antecedent was "params." If I'm right, then I'd go with

```suggestion
    // If the layer's partial route has params, the route is stored in layer.route. Otherwise, it's in layer.path
```

Reply:
Applied your suggestions and reworded the explanation slightly